monitoring/bin/deploy_monitoring_cluster.sh

```diff
@@ -114,16 +114,25 @@ if [ "$TLS_ENABLE" == "true" ]; then
   kubectl label cm -n $MON_NS node-exporter-tls-web-config sas.com/monitoring-base=kube-viya-monitoring
 fi
 
-log_info "User response file: [$PROM_OPER_USER_YAML]"
-log_info "Deploying the Kube Prometheus Stack. This may take a few minutes..."
-log_info "Installing via Helm...($(date) - timeout 20m)"
 if helm3ReleaseExists prometheus-operator $MON_NS; then
   promRelease=prometheus-operator
   promName=prometheus-operator
 else
   promRelease=v4m-prometheus-operator
   promName=v4m
 fi
+log_info "User response file: [$PROM_OPER_USER_YAML]"
+log_info "Deploying the Kube Prometheus Stack. This may take a few minutes..."
+if helm3ReleaseExists $promRelease $MON_NS; then
+  log_info "Upgrading via Helm...($(date) - timeout 20m)"
+else
+  grafanaPwd="$GRAFANA_ADMIN_PASSWORD"
+  if [ "$grafanaPwd" == "" ]; then
+    showPass="true"
+    grafanaPwd="$(date +%s | sha256sum | base64 | head -c 32 ; echo)"
```

Comment:
I thought you were creating a bash function (in common.sh?) for generating the password so we'd handle things consistently b/w logging and monitoring.

Reply:
Yep...coming in next commit